### DISCUSSION TITLE
lost pinned tabs after running updates

### DISCUSSION THREAD

**Discussion by pbwn** [Jelmerro/Vieb#555] [General]
I'm using endeavouros (arch) and ran updates. I'm on version 10.3.0 of vieb, so maybe it wasn't updated when i updated my system (I see the most current version is 11).

After rebooting and opening vieb I have all my tabs except the pinned tabs which open as empty tabs showing my bookmarks list. I have never experienced this before, it always successfully restores my session including pinned tabs

My question is whether there is a way to find my pinned tabs. I checked the files in .config 'hist' and 'tabs'. Is there some other way i might retrieve them? 

thanks

**Comment by Jelmerro** [comment]
I think what happened, is that the tabs failed to load in the previous run, and if you then restart the url is not saved as the "current" page for those tabs was still the newtab page, hence that being saved instead of the actual pages. I have had similar issues in the past, but only if I opened Vieb and closed it within seconds, so that none of the pages had time to load, hence all becoming newtab pages on the next start. I am sorry to say there is no option to recover them, except for finding the urls in the history page, since there is no history stored for open tabs, just the current state, which is now filled with empty tabs. I might try to look into this again, but have tried and failed to fix this in the past sadly. It's just very difficult to save something else than the current tab url conditionally while it is loading, as there are no built-in APIs in Electron to get the "next" url while it is still loading, so a custom solution for that is needed, which is there already at the moment, but loading pages usually save the "previous" url with that system, which is a little bit sad if the previous page was the newtab page. Sorry for the long and unsatisfying answer, but please do let me know if you have this again in the future, it might need another look in that case.

**Comment by pbwn** [comment]
Thanks a lot for your reply. I think you are exactly right in my case, ie I "opened Vieb and closed it within seconds, so that none of the pages had time to load". I open email in Vieb directly from neomutt when it contains html. Likely I saw the email was unimportant and closed Vieb before tabs loaded. Getting a more powerful computer will solve this for me; I only close Vieb because I have a very limited laptop and the Ram is consumed easily. I would be happy to leave it open.
